client/src/components/project/ComponentEditOverlay.js

```diff
@@ -195,72 +195,85 @@ const handleTextareaChange = e => {
           <Grid item xs={11}>
             <br></br>
             <Grid item sm={12} xs={6}>
-                <InputLabel fullWidth shrink required={true}>
-                    Component Name
-                </InputLabel>
-                <TextField  
-                    id="outlined-required"
-                    variant="outlined"
-                    style={{ width: 300 }}
-                    value={props.component.name}
-                    onChange={handleTextFieldChange}
-                />
+              <InputLabel fullWidth shrink required={true}>
+                Component Name
+              </InputLabel>
+              <TextField
+                id="outlined-required"
+                variant="outlined"
+                style={{ width: 300 }}
+                value={props.component.name}
+                onChange={handleTextFieldChange}
+              />
             </Grid>
             <br></br>
-            <Divider/>
+            <Divider />
             <br></br>
             <InputLabel fullWidth shrink required={true}>
-               Description
-              </InputLabel>
-            <TextareaAutosize onChange={handleTextareaChange}
-            value={props.component.description}
-            style={{ width: 600 , height: 100 ,"resize": "none"}}
-            maxRows={4}
+              Description
+            </InputLabel>
+            <TextareaAutosize
+              onChange={handleTextareaChange}
+              value={props.component.description}
+              style={{ width: 600, height: 100, resize: "none" }}
+              maxRows={4}
             />
             <br></br>
             <Grid item sm={12} xs={6}></Grid>
             <br></br>
 
             <Grid item sm={12} xs={6}>
-                <InputLabel fullWidth shrink required={true}>
-                  Component Lead
-                </InputLabel>
-                {selectedUser && <Autocomplete
-                  disabled
+              <InputLabel fullWidth shrink required={true}>
+                Component Lead
+              </InputLabel>
+              {props.component.name && ((props.component.user_id && selectedUser) || (!props.component.user_id)) && (
+                <Autocomplete
                   id="combo-box-demo"
                   options={users}
                   getOptionLabel={(option) => option.username}
                   style={{ width: 300 }}
-                  renderInput={(params) => <TextField {...params} variant="outlined" />}
+                  renderInput={(params) => (
+                    <TextField {...params} variant="outlined" />
+                  )}
                   value={selectedUser}
```

Comment:
We should use defaultValue to indicate the selected user if present. By passing null to defaultValue, we tell the component no values are selected